src/Components/Components/src/DefaultComponentActivator.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Default implementation of <see cref="IComponentActivator"/>.
+    /// </summary>
+    public class DefaultComponentActivator : IComponentActivator
+    {
+        // If no IComponentActivator is supplied by DI, the renderer uses this instance.
+        // It's internal because in the future, we might want to add per-scope state and then
+        // it would no longer be applicable to have a shared instance.
+        internal static IComponentActivator Instance { get; } = new DefaultComponentActivator();
+
+        /// <inheritdoc />
+        public IComponent CreateInstance(Type componentType)
+        {
+            var instance = Activator.CreateInstance(componentType);
```

Comment:
Unrelated to your change, but it might be worth profiling this to see if caching the factories makes a difference.

Reply:
We only create one per Renderer so it seems very unlikely to make a difference to anything. Was there a particular reason you’re suspicious it might have an observable effect?

Reply:
Oh you mean some kind of cached activator? I’m not even sure that’s possible without Reflection.Emit (which works but is slow on WebAssembly). Do you know of a good way to implement it?

Reply:
I was going to suggest ActivatorUtiltiies.CreateFactory but that uses Reflection.Emit. I guess once levi's ObjectFactory PR lands, we can revisit. Oh well 😞

Reply:
It's probably possible to do it like this:

```cs
internal interface ICachedActivator
{
    object CreateInstance();
}

internal class CachedActivator<T> : ICachedActivator where T: class, new()
{
    public object CreateInstance() => new T();
}
```

... and then separately create a `Dictionary<Type, ICachedActivator>` where for each `Type` we construct an instance one time using reflection.

However I have absolutely no reason to think this would be faster than `Activator.CreateInstance` when running on an interpreter. It's probably slower. With the interpreter, the only difference between `Activator.CreateInstance(typeof(T))` and `new T()` is having to look up the default constructor for `T`, which most likely takes less time than finding `typeof(T)` in a dictionary keyed on that. I don't know how `ObjectFactory` is implemented but it's hard to see how it would be faster than `Activator.CreateInstance` on an interpreter.

Reply:
I think `new T()` uses Activator.CreateInstance under the covers: https://devblogs.microsoft.com/premier-developer/dissecting-the-new-constraint-in-c-a-perfect-example-of-a-leaky-abstraction/.